pandas/tests/io/json/test_pandas.py

```diff
@@ -183,25 +167,21 @@ def test_roundtrip_str_axes(self, orient, convert_axes, numpy, dtype):
     @pytest.mark.parametrize("convert_axes", [True, False])
     @pytest.mark.parametrize("numpy", [True, False])
     def test_roundtrip_categorical(self, orient, convert_axes, numpy):
-        # TODO: create a better frame to test with and improve coverage
-        if orient in ("index", "columns"):
-            pytest.xfail(f"Can't have duplicate index values for orient '{orient}')")
+        cats = ["a", "b"]
+        df = pd.DataFrame(
+            pd.Categorical(cats), index=pd.CategoricalIndex(cats), columns=["cat"]
+        )
 
-        data = self.categorical.to_json(orient=orient)
-        if numpy and orient in ("records", "values"):
+        data = df.to_json(orient=orient)
+        if numpy and orient != "split":
```

Comment:
I removed the xfail preceding this because it was no longer applicable, but it unearthed a new issue when provided with any orient except split.

That keyword is deprecated in #28512  and scheduled for removal in 2.0, so figured not worth investing much in